cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -170,7 +170,7 @@ func (m *azureCache) regenerate() error {
 		if err != nil {
 			return err
 		}
-		klog.V(4).Infof("regenerate: found nodes for node group %s: %+v", ng.Id(), instances)
+		klog.V(4).Infof("regenerate: found %d nodes for node group %s: %+v", ng.Id(), len(instances), instances)
```

Comment:
Updated this to 1) not return early when there are no nodes and 2) updated the log message to include # of nodes found.